### PR TITLE
fix: adding credential check to gcp list clusters function

### PR DESCRIPTION
**Pull request by salonig23**
## Description

Add validate credential to the list clusters function to make sure the credentials are valid.

## Test Plan

Check if `det deploy gcp list --project-id <project_id>` is listing clusters without throwing any errors  

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: harness/determined/deploy/gcp/gcp.py [hunk 1]
```diff
@@ -322,27 +322,31 @@ def run_command(command: List[str], env: Dict[str, str], cwd: Optional[str] = No
     subprocess.check_call(command, env=env, stdout=sys.stdout, cwd=cwd)
 
 
-def validate_gcp_credentials(configs: Dict) -> None:
-    vars_file_path = get_terraform_vars_file_path(configs)
-    # Try to load google credentials from terraform vars when present.
-    if os.path.exists(vars_file_path):
-        tf_vars = terraform_read_variables(vars_file_path)
-        set_gcp_credentials_env(tf_vars)
+def validate_gcp_credentials(configs: Optional[Dict] = None, keypath: Optional[str] = None) -> None:
```
Comment (hamidzr) [inline harness/determined/deploy/gcp/gcp.py:325]: this is doing a couple of things that are not just validating gcp credentials. would describe it as a utility that loads and sets gcp credentials be more accurate?

Comment (hamidzr) [reply harness/determined/deploy/gcp/gcp.py:325]: could you please add a doc string that clarifies the inputs and the function's usage?

**Review by hamidzr — COMMENTED** [review]
could you please add tests steps? is there a related ticket that describes the bug?

**Comment by hamidzr** [comment]
I do get better error handling with this change

```
(determined) hbp ➜ determined git:(main) det deploy gcp list --project-id golabi
Traceback (most recent call last):
  File "/Users/hmd/projects/da/determined/harness/determined/cli/cli.py", line 230, in main
    parsed_args.func(parsed_args)
  File "/Users/hmd/projects/da/determined/harness/determined/deploy/gcp/cli.py", line 222, in handle_list
    return gcp.list_clusters(bucket_name)
  File "/Users/hmd/projects/da/determined/harness/determined/deploy/gcp/gcp.py", line 388, in list_clusters
    storage_client = storage.Client()
  File "/Users/hmd/.pyenv/versions/3.10.1/lib/python3.10/site-packages/google/cloud/storage/client.py", line 124, in __init__
    super(Client, self).__init__(
  File "/Users/hmd/.pyenv/versions/3.10.1/lib/python3.10/site-packages/google/cloud/client/__init__.py", line 318, in __init__
    _ClientProjectMixin.__init__(self, project=project, credentials=credentials)
  File "/Users/hmd/.pyenv/versions/3.10.1/lib/python3.10/site-packages/google/cloud/client/__init__.py", line 266, in __init__
    project = self._determine_default(project)
  File "/Users/hmd/.pyenv/versions/3.10.1/lib/python3.10/site-packages/google/cloud/client/__init__.py", line 285, in _determine_default
    return _determine_default_project(project)
  File "/Users/hmd/.pyenv/versions/3.10.1/lib/python3.10/site-packages/google/cloud/_helpers/__init__.py", line 154, in _determine_default_project
    _, project = google.auth.default()
  File "/Users/hmd/.pyenv/versions/3.10.1/lib/python3.10/site-packages/google/auth/_default.py", line 493, in default
    raise exceptions.DefaultCredentialsError(_HELP_MESSAGE)
google.auth.exceptions.DefaultCredentialsError: Could not automatically determine credentials. Please set GOOGLE_APPLICATION_CREDENTIALS or explicitly create credentials and re-run the application. For more information, please see https://cloud.google.com/docs/authentication/getting-started
Failed to list gcp cluster

(determined) hbp ➜ determined git:(main) gco -
Switched to branch 'fix_gcp_list'
Your branch is up to date with 'upstream/fix_gcp_list'.
(determined) hbp ➜ determined git:(fix_gcp_list) det deploy gcp list --project-id golabi
Unable to locate GCP credentials.Please setGOOGLE_APPLICATION_CREDENTIALSor explicitly create credentialsand re-run the application. For more information, please seehttps://docs.determined.ai/latest/sysadmin-deploy-on-gcp/install-gcp.html#credentials andhttps://cloud.google.com/docs/authentication/getting-started
```

the formatting is still a bit off though

File: harness/determined/deploy/gcp/gcp.py (hunk 1 above)
Comment (salonig23) [reply harness/determined/deploy/gcp/gcp.py]: Done!

**Comment by salonig23** [comment]
> Is there a related ticket that describes the bug?

Here: https://github.com/orgs/determined-ai/projects/13/views/1?pane=issue&itemId=31516926 

Not sure how to link it to the PR though

File: harness/determined/deploy/gcp/gcp.py (hunk 1 above)
Comment (hamidzr) [reply harness/determined/deploy/gcp/gcp.py]: thankyou

**Comment by hamidzr** [comment]
> > Is there a related ticket that describes the bug?
> 
> Here: https://github.com/orgs/determined-ai/projects/13/views/1?pane=issue&itemId=31516926
> 
> Not sure how to link it to the PR though

had a quick chat with @salonig23 it looks like 
@garrett361 had initially ran into these two issues:

```
$ det deploy gcp list --project-id determined-ai
    raise EnvironmentError(
OSError: Project was not passed and could not be determined from the environment.
Failed to list gcp cluster


And then after I pushed my initial changes he was getting this error:
det deploy gcp list --project-id determined-ai
    raise exceptions.RefreshError(
google.auth.exceptions.RefreshError: ('invalid_grant: Bad Request', {'error': 'invalid_grant', 'error_description': 'Bad Request'})
Failed to list gcp cluster
```

**Comment by salonig23** [comment]
Error for @coreystatendet before the PR: 
`(determined) hbp ➜ determined git:(fix_gcp_list) det deploy gcp list --project-id determined-ais
Traceback (most recent call last):
  File "/Users/hmd/projects/da/determined/harness/determined/cli/cli.py", line 230, in main
    parsed_args.func(parsed_args)
  File "/Users/hmd/projects/da/determined/harness/determined/deploy/gcp/cli.py", line 222, in handle_list
    return gcp.list_clusters(bucket_name, args.project_id)
  File "/Users/hmd/projects/da/determined/harness/determined/deploy/gcp/gcp.py", line 391, in list_clusters
    for blob in blobs:
  File "/Users/hmd/.pyenv/versions/3.10.1/lib/python3.10/site-packages/google/api_core/page_iterator.py", line 208, in _items_iter
    for page in self._page_iter(increment=False):
  File "/Users/hmd/.pyenv/versions/3.10.1/lib/python3.10/site-packages/google/api_core/page_iterator.py", line 244, in _page_iter
    page = self._next_page()
  File "/Users/hmd/.pyenv/versions/3.10.1/lib/python3.10/site-packages/google/api_core/page_iterator.py", line 373, in _next_page
    response = self._get_next_page_response()
  File "/Users/hmd/.pyenv/versions/3.10.1/lib/python3.10/site-packages/google/api_core/page_iterator.py", line 432, in _get_next_page_response
    return self.api_request(
  File "/Users/hmd/.pyenv/versions/3.10.1/lib/python3.10/site-packages/google/cloud/storage/_http.py", line 80, in api_request
    return call()
  File "/Users/hmd/.pyenv/versions/3.10.1/lib/python3.10/site-packages/google/api_core/retry.py", line 283, in retry_wrapped_func
    return retry_target(
  File "/Users/hmd/.pyenv/versions/3.10.1/lib/python3.10/site-packages/google/api_core/retry.py", line 190, in retry_target
    return target()
  File "/Users/hmd/.pyenv/versions/3.10.1/lib/python3.10/site-packages/google/cloud/_http/__init__.py", line 480, in api_request
    raise exceptions.from_http_response(response)
google.api_core.exceptions.NotFound: 404 GET https://storage.googleapis.com/storage/v1/b/determined-ais-determined-deploy/o?projection=noAcl&prettyPrint=false: The specified bucket does not exist.
Failed to list gcp cluster`

Error after initial changes:
`det deploy gcp list --project-id determined-ai
Traceback (most recent call last):
  File "/home/user/determined/corey/determined/harness/determined/cli/cli.py", line 230, in main
    parsed_args.func(parsed_args)
  File "/home/user/determined/corey/determined/harness/determined/deploy/gcp/cli.py", line 222, in handle_list
    return gcp.list_clusters(bucket_name, args.project_id)
  File "/home/user/determined/corey/determined/harness/determined/deploy/gcp/gcp.py", line 384, in list_clusters
    for blob in blobs:
  File "/home/user/determined/corey/determined/env/lib/python3.8/site-packages/google/api_core/page_iterator.py", line 208, in _items_iter
    for page in self._page_iter(increment=False):
  File "/home/user/determined/corey/determined/env/lib/python3.8/site-packages/google/api_core/page_iterator.py", line 244, in _page_iter
    page = self._next_page()
  File "/home/user/determined/corey/determined/env/lib/python3.8/site-packages/google/api_core/page_iterator.py", line 373, in _next_page
    response = self._get_next_page_response()
  File "/home/user/determined/corey/determined/env/lib/python3.8/site-packages/google/api_core/page_iterator.py", line 432, in _get_next_page_response
    return self.api_request(
  File "/home/user/determined/corey/determined/env/lib/python3.8/site-packages/google/cloud/storage/_http.py", line 72, in api_request
    return call()
  File "/home/user/determined/corey/determined/env/lib/python3.8/site-packages/google/api_core/retry.py", line 349, in retry_wrapped_func
    return retry_target(
  File "/home/user/determined/corey/determined/env/lib/python3.8/site-packages/google/api_core/retry.py", line 191, in retry_target
    return target()
  File "/home/user/determined/corey/determined/env/lib/python3.8/site-packages/google/cloud/_http/__init__.py", line 482, in api_request
    response = self._make_request(
  File "/home/user/determined/corey/determined/env/lib/python3.8/site-packages/google/cloud/_http/__init__.py", line 341, in _make_request
    return self._do_request(
  File "/home/user/determined/corey/determined/env/lib/python3.8/site-packages/google/cloud/_http/__init__.py", line 379, in _do_request
    return self.http.request(
  File "/home/user/determined/corey/determined/env/lib/python3.8/site-packages/google/auth/transport/requests.py", line 545, in request
    self.credentials.before_request(auth_request, method, url, request_headers)
  File "/home/user/determined/corey/determined/env/lib/python3.8/site-packages/google/auth/credentials.py", line 151, in before_request
    self.refresh(request)
  File "/home/user/determined/corey/determined/env/lib/python3.8/site-packages/google/oauth2/credentials.py", line 339, in refresh
    ) = reauth.refresh_grant(
  File "/home/user/determined/corey/determined/env/lib/python3.8/site-packages/google/oauth2/reauth.py", line 367, in refresh_grant
    _client._handle_error_response(response_data, retryable_error)
  File "/home/user/determined/corey/determined/env/lib/python3.8/site-packages/google/oauth2/_client.py", line 74, in _handle_error_response
    raise exceptions.RefreshError(
google.auth.exceptions.RefreshError: ('invalid_grant: Bad Request', {'error': 'invalid_grant', 'error_description': 'Bad Request'})
Failed to list gcp cluster`

**Review by hamidzr — APPROVED** [review]
the minor formatting issue I saw looks great. although probably not the goal of the pr I saw our error handling the case of a 404 project-id isn't the best IMO we don't need to resolve this to land the fix for the release (higher risk and time). Let's ask @coreystatendet  to give it a try too

```
(determined) hbp ➜ determined git:(fix_gcp_list) det deploy gcp list --project-id determined-ais
Traceback (most recent call last):
  File "/Users/hmd/projects/da/determined/harness/determined/cli/cli.py", line 230, in main
    parsed_args.func(parsed_args)
  File "/Users/hmd/projects/da/determined/harness/determined/deploy/gcp/cli.py", line 222, in handle_list
    return gcp.list_clusters(bucket_name, args.project_id)
  File "/Users/hmd/projects/da/determined/harness/determined/deploy/gcp/gcp.py", line 391, in list_clusters
    for blob in blobs:
  File "/Users/hmd/.pyenv/versions/3.10.1/lib/python3.10/site-packages/google/api_core/page_iterator.py", line 208, in _items_iter
    for page in self._page_iter(increment=False):
  File "/Users/hmd/.pyenv/versions/3.10.1/lib/python3.10/site-packages/google/api_core/page_iterator.py", line 244, in _page_iter
    page = self._next_page()
  File "/Users/hmd/.pyenv/versions/3.10.1/lib/python3.10/site-packages/google/api_core/page_iterator.py", line 373, in _next_page
    response = self._get_next_page_response()
  File "/Users/hmd/.pyenv/versions/3.10.1/lib/python3.10/site-packages/google/api_core/page_iterator.py", line 432, in _get_next_page_response
    return self.api_request(
  File "/Users/hmd/.pyenv/versions/3.10.1/lib/python3.10/site-packages/google/cloud/storage/_http.py", line 80, in api_request
    return call()
  File "/Users/hmd/.pyenv/versions/3.10.1/lib/python3.10/site-packages/google/api_core/retry.py", line 283, in retry_wrapped_func
    return retry_target(
  File "/Users/hmd/.pyenv/versions/3.10.1/lib/python3.10/site-packages/google/api_core/retry.py", line 190, in retry_target
    return target()
  File "/Users/hmd/.pyenv/versions/3.10.1/lib/python3.10/site-packages/google/cloud/_http/__init__.py", line 480, in api_request
    raise exceptions.from_http_response(response)
google.api_core.exceptions.NotFound: 404 GET https://storage.googleapis.com/storage/v1/b/determined-ais-determined-deploy/o?projection=noAcl&prettyPrint=false: The specified bucket does not exist.
Failed to list gcp cluster
```

**Comment by hamidzr** [comment]
that second error could have been any of the following I think

Token Expiry
Time Synchronization Issue
Incorrect/Changed Credentials
Refresh Token Limit Reached

**Pull request merged**